snorkel/labeling/model/sparse_label_model/base_sparse_label_model.py

```diff
@@ -0,0 +1,223 @@
+# -*- coding: utf-8 -*-
+"""Sparse Data Helpers
+
+Indexing throughout this module is 0 based, with the assumption that "abstains" are ommited.
+
+When working with larger datasets, it can be convenient to load the data in sparse format. This module
+provides utilities to do so. We provide functions for a number of cases.
+
+The user has the AugmentedMatrix (L_ind) in tuple form. AugmentedMatrix is of shape (num_examples,numfuncs*num_classes)
```

Comment:
Nit: Let's specify each of these as "Case 1", "Case 2", etc. so it's both (a) more obvious that these are mutually exclusive, and (b) easier to reference.